Cython/Compiler/Parsing.py

```diff
@@ -2337,14 +2338,25 @@ def p_statement(s, ctx, first_statement = 0):
 
 def p_statement_list(s, ctx, first_statement = 0):
     # Parse a series of statements separated by newlines.
+    global stat
     pos = s.position()
     stats = []
+
     while s.sy not in ('DEDENT', 'EOF'):
-        stat = p_statement(s, ctx, first_statement = first_statement)
+        try:
+            stat = p_statement(s, ctx, first_statement = first_statement)
+        except Exception as exc:
+            if s.fault_tolerant == True:
+                report_error(CompileError(pos, "Internal Error: "+ str(exc)))
```

Comment:
We should also check if it's not already a `CompileError`, just in case.

i.e.:
```
if not isinstance(exc, CompileError):
    exc = CompileError(pos, "Internal Error: "+ str(exc)
report_error(exc)
```